sktime/transformations/series/difference.py

```diff
@@ -121,16 +126,51 @@ class Differencer(BaseTransformer):
         "capability:inverse_transform": True,
     }
 
-    def __init__(self, lags=1, drop_na=True):
+    VALID_NA_HANDLING_STR = ["drop_na", "keep_na", "fill_zero"]
+
+    def __init__(self, lags=1, drop_na=None, na_handling="drop_na"):
         self.lags = lags
         self.drop_na = drop_na
+        self.na_handling = self._check_na_handling(na_handling)
+        # note: internally, we will use self._na_handling
+        #   because we must never change input param saves for sklearn compatibility
+        #   and because we need to "translate" the old "drop_na" arg
+        #   this could, in certain cases, overwrite the self. parameter
+        #   and cause sklearn compatibility issues due to the point mentioned
+
+        translate_arg = {True: "drop_na", False: "keep_na"}
+
+        if drop_na is not None:
+            warn(
+                f"the drop_na parameter is deprecated and will be removed in 0.13.0, "
+                f'use na_handling="{translate_arg[drop_na]}" instead'
+            )
```

Comment:
should we not use `DeprecationWarning`? We might have to add that to our deprecation doc as well

Reply:
yes, you are right. Might not be too clear in the deprecation section what one needs to do if the decorator cannot be used.

Reply:
added the `DeprecationWarning`